armi/reactor/blocks.py

```diff
@@ -1519,6 +1533,58 @@ def coords(self, rotationDegreesCCW=0.0):
             round(y, units.FLOAT_DIMENSION_DECIMALS),
         )
 
+    def createCopy(self):
+        """
+        Create a blank copy of a block (ligher weight than deepcopy)
+
+        Notes
+        -----
+        If you make a new block, you must add it to an assembly and a reactor.
+        """
+
+        b = self.__class__(self.getName(), height=self.getHeight())
+        b.setType(self.getType())
+
+        # assign macros and LFP
+        b.macros = self.macros
+        b._lumpedFissionProducts = self._lumpedFissionProducts
+        b.p.buGroup = self.p.buGroup
+
+        hexComponent = Hexagon(
```

Comment:
Sorry, stupid question, does this work for non-hex blocks?

Because 2/3 of our users don't have hex blocks.

Reply:
I think that this is addressed on the general block as just a deepcopy now.

Reply:
That's correct @jhader. A faster/lighter copy function could be written for Cartesian or other blocks and it would probably look quite similar to this one, but for now I'm trying to minimize the scope of the change.

I don't have enough knowledge the use cases for other blocks to write a good `createCopy` that generalizes to all block types, but someone who uses another block type frequently should be able to implement a function modeled on this one without much trouble.